pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1324,47 +1324,80 @@ public IdealState apply(@Nullable IdealState idealState) {
         }, RetryPolicies.exponentialBackoffRetryPolicy(5, 500L, 2.0f));
   }
 
-  public void deleteOfflineTable(String tableName) {
-    String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(tableName);
+  public void deleteOfflineTable(String rawTableName) {
+    String offlineTableName = TableNameBuilder.OFFLINE.tableNameWithType(rawTableName);
+    LOGGER.info("Deleting table {}: Start", offlineTableName);
+    long startTime = System.currentTimeMillis();
 
     // Remove the table from brokerResource
     HelixHelper.removeResourceFromBrokerIdealState(_helixZkManager, offlineTableName);
+    long removeFromBrokerResourceFinishTime = System.currentTimeMillis();
+    LOGGER.info("Deleting table {}: Removed from broker resource in {}ms", offlineTableName,
+        (removeFromBrokerResourceFinishTime - startTime));
 
     // Drop the table
     if (_helixAdmin.getResourcesInCluster(_helixClusterName).contains(offlineTableName)) {
       _helixAdmin.dropResource(_helixClusterName, offlineTableName);
     }
+    long dropTableFinishTime = System.currentTimeMillis();
+    LOGGER.info("Deleting table {}: Dropped table in {}ms", offlineTableName,
+        (dropTableFinishTime - removeFromBrokerResourceFinishTime));
 
     // Remove all segments for the table
     _segmentDeletionManager.removeSegmentsFromStore(offlineTableName, getSegmentsFor(offlineTableName));
     ZKMetadataProvider.removeResourceSegmentsFromPropertyStore(_propertyStore, offlineTableName);
+    long removeSegmentFinishTime = System.currentTimeMillis();
+    LOGGER.info("Deleting table {}: Removed segments in {}ms", offlineTableName,
```

Comment:
Removed segment metadata

Reply:
Actually it removes metadata from ZK AND moves segment files to `Delete_Segment` dir as well. That's why I name it to `Removed segments`

Reply:
ah, ok. Then call it as "Removed segment metadata and stored segments" Maybe we need indepdent steps to determine which part is taking time?

Reply:
Suggestion accepted. 👍